### PR TITLE
Local Scaling Methods & Rotation Transform (Torch)

### PR DESCRIPTION
**Pull request by marxprop**
Added Random Rotation with a default angle of (15deg) which applies rotation between (-15 and 15 degrees)  on randomly selected samples.

Added new scaling methods to Pytorch  which include: 
- Global Normalization
- Local Normalization,
- Local Scaling and
-  MinMax Scaler

All the scaling methods except from Global Normalizing are done online for each image while training. It is applied after all data augmentation transforms, if any.  

Global Normalization is set as the default to `dgbpy.mlapply.getScaledTrainingData()` using the global mean and std of all the samples for standard scaling.

The Global Normalization method is applied to all workflows with Log output. `dgbpy.hdf5.isLogOutput()`

Information about the scaling method and its property is also saved to any trained model.

Increased the default number of Unet blocks in Pytorch from 1 to 2 so models can learn better.

Remove dict support for pytorch transforms. UI now uses default transform parameters available in `dgbpy.dgbtorch.torch_dict`.

### CODE REVIEW

File: dgbpy/torch_classes.py [hunk 1]
```diff
@@ -746,82 +745,170 @@ def transform(self, arr):
...
+    import cv2
+  except ModuleNotFoundError:
+    return False
+  return True
+
+class RandomRotation():
+    def __init__(self, p=0.3, angle=15):
+        self.p = p
+        self.angle = angle
+        import cv2
```
Comment (waynegm) [inline dgbpy/torch_classes.py:776]: Would it be possible to avoid adding a dependency on OpenCV by using scipy.ndimage which is already included in the OpendTect Python distribution? There is scipy.ndimage.rotate - don't know if it does the same.
